src/microbe_stage/FloatingChunk.cs

```diff
@@ -262,19 +275,13 @@ public override void _Process(float delta)
             }
 
             if (DeleteOnTouch || disappear)
-            {
-                if (Dissolves)
-                {
-                    isDissolving = true;
-                }
-                else
-                {
-                    this.DetachAndQueueFree();
-                }
+                DissolveOrRemove();
 
-                break;
-            }
+            break;
```

Comment:
this is wrong logic wise as far as I see it, this always breaks on the first iteration. This break belongs into the if two lines above. Add brackets.